### PR TITLE
Separating coap code from mbed-client-c

### PR DESCRIPTION
**Pull request by yogpan01**
This commit includes
 - Removing coap header and source files from the repository and moving them into their own repo https://github.com/ARMmbed/libcoap

### CODE REVIEW

**Comment by yogpan01** [comment]
@kjbracey-arm FYI

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# Makefile for combined NSDL+COAP library
+# Makefile for combined NSDL library
```
Comment (kjbracey-arm) [inline Makefile]: It's not "combined" any more

File: Makefile [hunk 2]
```diff
@@ -19,18 +19,15 @@ LIB = libnsdl.a
...
 
 override CFLAGS += -DVERSION='"$(VERSION)"'
 
 override CFLAGS += -Isource/libNsdl/src/include/
-override CFLAGS += -Isource/libCoap/src/include/
 SERVLIB_DIR := ../libService
 override CFLAGS += -I$(SERVLIB_DIR)/libService
 override CFLAGS += -Insdl-c/
+override CFLAGS += -I../libcoap/libcoap/
+override CFLAGS += -I../libcoap/source/include/
```
Comment (kjbracey-arm) [inline Makefile]: Do you really need to include /source/include? If you do, you're cheating a bit - as they should be private headers, not accessed from outside the library.

I guess you could argue that this component is coupled in some way. Would only cause a problem if you for some reason had binary coap library and were trying to compile source nsdl library.

File: module.json [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
-    "source/libNsdl/src/include",
-    "source/libCoap/src/include"
+    "source/libNsdl/src/include"
   ],
   "dependencies": {
     "nanostack-libservice": "^3.0.0",
     "mbed-trace": ">=0.2.0,<2.0.0",
-    "nanostack-randlib": "^1.2.0"
+    "nanostack-randlib": "^1.2.0",
+    "libcoap": "^1.0.0"
```
Comment (kjbracey-arm) [inline module.json]: As per discussion, libcoap versioning should really continue from the same point as nsdl, so 5.0.0.

**Comment by kjbracey-arm** [comment]
This is inconsistent with #121, in that you flatten the directory to just source/ there, but this remains as source/libNsdl.

File: Makefile (hunk 2 above)
Comment (yogpan01) [inline Makefile]: Unfortunately sn_grs has some internal data handling with coap structure , hence it needs these private headers. 
Is it ok, if we remove that dependency as a separate task, once we have this new structure in place ?

File: module.json (hunk 3 above)
Comment (yogpan01) [inline module.json]: Agreed.

File: Makefile (hunk 2 above)
Comment (kjbracey-arm) [inline Makefile]: Sure, it can be tidied up later.

**Comment by yogpan01** [comment]
Abandoning this PR and superseding with https://github.com/ARMmbed/mbed-client-c/pull/125

**Pull request closed without merging**